File: Presentation/Contracts/Common/ValueObjects/Address.cs
﻿using AutoMapper;

namespace Presentation.Contracts.Common.ValueObjects
{
    public class AddressAdapter : Profile
    {
        public AddressAdapter() 
        {
            CreateMap<Address, Domain.ValueObjects.Address>().ForMember(dest => dest.GpsCoordinate, opt =>
            {
                opt.Condition(src => src.GpsCoordinate != null && (src.GpsCoordinate.Latitude != null || src.GpsCoordinate.Longitude != null));
            });
            CreateMap<Address, Domain.ValueObjects.Address>().ReverseMap();
        }
    }
    public class Address
    {
        public string Street { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string PostalCode { get; set; }
        public string Country { get; set; }
        public GpsCoordinate? GpsCoordinate { get; set; }

        public Address(string street, string city, string state, string postalCode, string country, GpsCoordinate? gpsCoordinate = null)
        {
            Street = street;
            City = city;
            State = state;
            PostalCode = postalCode;
            Country = country;
            GpsCoordinate = gpsCoordinate;
        }

        public Address()
        {
        }
    }
}

File: Python script for generating transportations/address.py
class Address:
    valid : bool = False
    street = ''
    city = ''
    state = ''
    postal_code = ''
    country = ''
    latitude : float = 0
    longitude : float = 0

    def isValid(self):
        self.valid = self.street != '' and self.city != '' and self.state != '' and self.postal_code != '' and self.country != '' and (self.latitude != 0 or self.latitude != None) and (self.longitude != 0 or self.longitude != None)